pkg/kn/flags/podspec_helper.go

```diff
@@ -638,3 +663,22 @@ func reviseVolumesToRemove(volumeMounts []corev1.VolumeMount, volumesToRemove []
 	}
 	return volumesToRemove
 }
+
+func decodeContainersFromFile(filename string) (*corev1.PodSpec, error) {
+	var f *os.File
+	var err error
+	if filename == "-" {
+		f = os.Stdin
+	} else {
+		f, err = os.Open(filename)
+		if err != nil {
+			return nil, err
+		}
+	}
+	podSpec := &corev1.PodSpec{}
+	decoder := yaml.NewYAMLOrJSONDecoder(f, 512)
+	if err = decoder.Decode(podSpec); err != nil {
+		return nil, err
+	}
+	return podSpec, f.Close()
```

Comment:
Not sure if `Stdin` should be closed and what side-effect that might have. tbh, an error when closing a file when reading only from it could be also ignored imo